lib/ansible/modules/remote_management/dellemc/ome/dellemc_ome_device_facts.py

```diff
@@ -0,0 +1,418 @@
+                 type: list
+            device_service_tag:
+                 description:
+                    - A list of service tags are applicable for C(detailed_inventory)
+                      and C(subsystem_health).
+                 type: list
+            inventory_type:
+                description:
+                    - For C(detailed_inventory), it returns details of the specified inventory type.
+                type: str
+            filter:
+                description:
+                    - For C(basic_inventory), it filters the collection of devices.
+                      I(filter) query format should be aligned with OData standards.
+                type: str
+
+requirements:
+    - "python >= 2.7.5"
+author: "Sajna Shetty(@Sajna-Shetty)"
+'''
+
+EXAMPLES = """
+---
+- name: Retrieve basic inventory of all devices.
+  dellemc_ome_device_facts:
+    hostname: "192.168.0.1"
+    username: "username"
+    password: "password"
+
+- name: Retrieve basic inventory for devices identified by IDs 33333 or 11111 using filtering.
+  dellemc_ome_device_facts:
+    hostname: "192.168.0.1"
+    username: "username"
+    password: "password"
+    fact_subset: "basic_inventory"
+    system_query_options:
+      filter: "Id eq 33333 or Id eq 11111"
+
+- name: Retrieve inventory details of specified devices identified by IDs 11111 and 22222.
+  dellemc_ome_device_facts:
+    hostname: "192.168.0.1"
+    username: "username"
+    password: "password"
+    fact_subset: "detailed_inventory"
+    system_query_options:
+      device_id:
+        - 11111
+        - 22222
+
+- name: Retrieve inventory details of specified devices identified by service tags MXL1234 and MXL4567.
+  dellemc_ome_device_facts:
+    hostname: "192.168.0.1"
+    username: "username"
+    password: "password"
+    fact_subset: "detailed_inventory"
+    system_query_options:
+      device_service_tag:
+        - MXL1234
+        - MXL4567
+
+- name: Retrieve details of specified inventory type of specified devices identified by ID and service tags.
+  dellemc_ome_device_facts:
+    hostname: "192.168.0.1"
+    username: "username"
+    password: "password"
+    fact_subset: "detailed_inventory"
+    system_query_options:
+      device_id:
+        - 11111
+      device_service_tag:
+        - MXL1234
+        - MXL4567
+      inventory_type: "serverDeviceCards"
+
+- name: Retrieve subsystem health of specified devices identified by service tags.
+  dellemc_ome_device_facts:
+    hostname: "192.168.0.1"
+    username: "username"
+    password: "password"
+    fact_subset: "subsystem_health"
+    system_query_options:
+      device_service_tag:
+        - MXL1234
+        - MXL4567
+
+"""
+
+RETURN = """
+---
+msg:
+  type: str
+  description: Over all device_facts status.
+  returned: on error
+  sample: "Failed to fetch the device facts"
+ansible_facts:
+  type: dict
+  description: Device inventory details.
```

Comment:
I don't think it is clear from this description under which name the result is stored in `ansible_facts`. Maybe you should add an example which prints out the returned facts.

Reply:
Thanks for the review. I have updated the description and example part.

Reply:
@felixfontein As we are connecting to remote system, we are retrieving unique information like network address,mac address and hardware details. As per [ansible doc](https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#variables-discovered-from-systems-facts), facts are used when connecting to Remote system,

So is not it better to have it as `_facts` module?

Reply:
I don't know. I can think of reasons why it should be `_facts`, and reasons why it should be `_info`. I guess this is a question for the public Ansible meeting...

Reply:
Discussion from meeting: should be `_facts`

Reply:
Thank you